src/posts/interning-at-khan-academy.md

```diff
@@ -0,0 +1,51 @@
+
+At Khan Academy, we use React.js, something I had worked with before, but only in the context of [Scala.js](https://www.scala-js.org) applications. As part of making changes to the math keypad, I had to learn how to write idiomatic ES6 code. Working on Khan Academy gave me a great opportunity to see use of React with ES6 in a production environment, and left me with many ideas on how React.js could be used more seamlessly with Scala.js. In addition to learning React and ES6, I got a chance to work with the plethora of open-source libraries at Khan Academy such as Aphrodite ([https://github.com/khan/aphrodite](https://github.com/khan/aphrodite)) and KaTeX ([https://github.com/khan/katex](https://github.com/khan/katex)).
+
+![changes to the math keypad](/images/interning-at-khan-academy/image_0.png)
+
+*an early screenshot of changes to the math keypad*
+
+# Part 2: iframes, iframes, iframes
+
+Now that I was used to the flow of picking a task, creating a code review, and deploying my changes, I started on my first project -- improving our content editor to be mobile first. This involved pulling together specs for what types of devices students use Khan Academy on, rendering content with the new styles the rest of the team was working on, isolating the rendered content to prevent styles leaking from the rest of the editor, and implementing touch emulation for interacting with the mobile previews.
+
+To implement the isolation of content, I decided to render previews inside of iframes. In addition to providing style isolation, iframes also made it possible to properly render components that are rendered directly to the body, such as the math keypad I worked on earlier. To transfer data between the parent editor and the child iframe, the parent first stores a JS object representation of the exercise content in a global variable and then sends a message ([http://ejohn.org/blog/cross-window-messaging/](http://ejohn.org/blog/cross-window-messaging/)) to the iframe. When the iframe receives this message, it grabs the global JS object from the parent (because the iframe and parent are on the same domain this is allowed by the browser) and renders that content. With this implementation, preview updates happen immediately after content edits, and with a special `shouldComponentUpdate` iframe loads are minimized so that in most cases, the only loading delay happens during the initial editor load.
```

Comment:
You should explain why storing an object as a global is necessary.  Why can't we just serialize the object and send it using postMessage like normal?

Maybe "... and then sends a message using [window.postMessage](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) to the iframe."

Readers might also be interested in the special `shouldComponentUpdate`.

This paragraph should be split up.